File: SocketRocket/Internal/Utilities/Compat.m
//
//  Compat.m
//  SocketRocket
//
//  Created by kbinani on 2019/05/08.
//

#import <Foundation/Foundation.h>
#import <SocketRocket/SocketRocket.h>
#import "SocketRocket-Swift.h"

static void compat_SRSecurityPolicy() {
    [[SRSecurityPolicy alloc] init];
    [[SRSecurityPolicy alloc] initWithCertificateChainValidationEnabled:YES];
    SRSecurityPolicy *policy = [SRSecurityPolicy defaultPolicy];
    [SRSecurityPolicy pinnningPolicyWithCertificates: NSArray.new];
    NSInputStream *stream = [[NSInputStream alloc] initWithFileAtPath:@""];
    [policy updateSecurityOptionsInStream:stream];
    SecTrustRef trust = NULL;
    SecTrustCreateWithCertificates(CFArrayCreate(NULL, NULL, 0, NULL), NULL, &trust);
    [policy evaluateServerTrust:trust forDomain:@""];
}

File: SocketRocket/Internal/Utilities/Compat.swift
private func compat_SRSecurityPolicy() {
    do {
        SRSecurityPolicy.init()
        SRSecurityPolicy.init(certificateChainValidationEnabled: true)
        let policy = SRSecurityPolicy.default()
        SRSecurityPolicy.pinnningPolicy(withCertificates: [])
        let stream = InputStream(fileAtPath: "")!
        policy.updateSecurityOptions(in: stream)
        var trust: SecTrust? = nil
        let certs = [AnyObject]()
        SecTrustCreateWithCertificates(certs as! CFArray, nil, &trust)
        policy.evaluateServerTrust(trust!, forDomain: "")
    }
}
